### PR TITLE
Remove insecure md5 example

### LINKED ISSUE #13: Remove insecure MD5 from documentation
It tears my soul that documentation about authentication mentions md5 so much.
There is even an example how to use Autheticatin with md5.
MD5 is insecure and todays RDBS support newer and secure methods....
(There is also md5 note in code comments)

https://docs.zendframework.com/zend-authentication/adapter/dbtable/credential-treatment/

### PR DESCRIPTION
**Pull request by Xerkus**
Fixes #13 

- [x] Is this related to documentation?
Old documentation for `CredentialTreatmentAdapter` provides `md5()` in its usage example which might prompt users to utilize insecure practices. This PR removes `md5()` usage from the documentation and adds further guidance towards `CallbackCheckAdapter` and PHP functions `password_hash()` and `password_verify()`

### CODE REVIEW

File: docs/book/adapter/dbtable/credential-treatment.md [hunk 1]
```diff
@@ -3,8 +3,30 @@
...
+rather use the RDBMS specific function such as
+[`PASSWORD(?)` for MySQL](http://dev.mysql.com/doc/refman/5.7/en/password-hashing.html) or
+[`crypt()` for PostgreSQL](https://www.postgresql.org/docs/11/pgcrypto.html#id-1.11.7.34.6).
+More details are available in the next section.
+
+## Security considerations
+
+Passing passwords to database in plaintext for insert or verification is
+generally not recommended.  
+Sql statements can and usually are logged by the database, passwords in them
```
Comment (weierophinney) [inline docs/book/adapter/dbtable/credential-treatment.md]: Add an `and` after the comma.

**Comment by Xerkus** [comment]
@tdutrion I cherry picked your commit fixing md5 usage in the docs. Thanks.

File: docs/book/adapter/dbtable/credential-treatment.md [hunk 2]
```diff
@@ -3,8 +3,30 @@
...
+become visible to anyone with access to the logs or monitoring tools that
+consume those logs.
+
+Safer way is to hash passwords and verify them against stored hash on the
+application side. This way password never needs to leave application and only
+hashed value exchanged with database.
```
Comment (weierophinney) [inline docs/book/adapter/dbtable/credential-treatment.md]: Rephrase to:

```markdown
The safer approach is to hash passwords, and to verify them against a stored hash in your application code. This way the password never leaves the application, and only the hashed value is exchanged with the database.
```

File: docs/book/adapter/dbtable/credential-treatment.md [hunk 3]
```diff
@@ -3,8 +3,30 @@
...
+
+As such, this adapter is not recommended for new applications and existing
```
Comment (weierophinney) [inline docs/book/adapter/dbtable/credential-treatment.md]: Add a comma after `applications`.

File: docs/book/adapter/dbtable/credential-treatment.md [hunk 4]
```diff
@@ -3,8 +3,30 @@
...
+applications should consider migrating to using PHP provided password handling
+functions `password_hash()` and `password_verify()`. See
```
Comment (weierophinney) [inline docs/book/adapter/dbtable/credential-treatment.md]: Replace starting at `using` and ending at `functions` with the following: `using PHP-provided password handling functions such as`

File: docs/book/adapter/dbtable/credential-treatment.md [hunk 5]
```diff
@@ -238,7 +260,9 @@ $sqlAlter = "ALTER TABLE [users] "
     . "AFTER [password]";
 ```
 
-Salts should be created *for each user* using a cryptographically sound pseudo-random number generator (CSPRNG). PHP 7 provides an implementation via `random_bytes`:
+Salts should be created *for each user* using a cryptographically sound pseudo-random number generator (CSPRNG).
+PHP 7 provides an implementation via `random_bytes` (and 
+[`random_compat` for older supported versions of PHP](https://github.com/paragonie/random_compat)):
```
Comment (weierophinney) [inline docs/book/adapter/dbtable/credential-treatment.md]: Rephrase this to `(and the [random_compat package provides them for older, supported versions of PHP](...))` (fill in the URL where I put ellipses).

**Review by weierophinney — CHANGES_REQUESTED** [review]
Great idea! I've added a few recommendations for changes, but otherwise, it looks good.

File: docs/book/adapter/dbtable/credential-treatment.md (hunk 5 above)
Comment (Xerkus) [reply docs/book/adapter/dbtable/credential-treatment.md]: Since this package targets ^5.6, those functions are already available. Should we mention compat?

Comment (Xerkus) [reply docs/book/adapter/dbtable/credential-treatment.md]: Oh, nvm. `random_bytes()` is ^7.0

**Review by weierophinney — APPROVED** [review]
